spec/system/frontend/solidus_stripe/checkout_spec.rb

```diff
@@ -2,218 +2,302 @@
-    expect(SolidusStripe::PaymentSource.last.stripe_payment_method_id).to be_present
-
-    # Pay with the newly created wallet source
-    visit_payment_step(user: user)
-    find_existing_payment_radio(user.wallet_payment_sources.first.id).choose
-    submit_payment
-    expect_payments_state(Spree::Order.last, ['checkout'])
-    confirm_order
-
-    order = Spree::Order.last
-    payment = order.payments.valid.first
-    expect(Spree::Order.count).to eq(2)
-    expect(order.user).to eq(user)
-    expect_checkout_completion(order)
-    expect_payments_state(order, ['pending'])
-    expect(order.payments.valid.count).to eq(1)
-    payment.capture!
-    expect_payments_state(order, ['completed'], outstanding: 0)
-    expect(SolidusStripe::PaymentSource.count).to eq(1)
-    expect(SolidusStripe::PaymentSource.last.stripe_payment_method_id).to be_present
-    expect(payment.source).to eq(reusable_source)
-  end
+  # To learn more about setup_future_usage in different contexts:
+  # - SetupIntents   : https://stripe.com/docs/payments/setup-intents#increasing-success-rate-by-specifying-usage
+  # - PaymentIntents : https://stripe.com/docs/payments/payment-intents#future-usage
+  ['', 'on_session', 'off_session'].each do |setup_future_usage|
+    context "with Stripe Setup Intent and setup_future_usage=#{setup_future_usage}" do
+      before do
+        create_stripe_payment_method(
+          intents_flow: 'setup',
+          setup_future_usage: setup_future_usage
+        )
+      end
 
-  it "completes as a guest using setup intents" do
-    create(:stripe_payment_method, preferred_stripe_intents_flow: 'setup')
-    visit_payment_step(user: nil)
-    choose_new_stripe_payment
-    fill_stripe_form
-    submit_payment
-    expect_payments_state(Spree::Order.last, ['checkout'])
-    confirm_order
-
-    order = Spree::Order.last
-    expect(Spree::Order.count).to eq(1)
-    expect_checkout_completion(order)
-    expect_payments_state(order, ['pending'], outstanding: order.total)
-    order.payments.first.capture!
-    expect_payments_state(order, ['completed'], outstanding: 0)
-  end
+      context 'with a registered user' do
+        it 'creates a setup intent and successfully processes payment' do
+          successfully_creates_a_setup_intent(user: create(:user))
+          complete_order
+          expect_payment_intent_to_be_created_successfully
+          capture_last_valid_payment
+        end
 
-  [true, false].each do |skip_confirm|
-    it "completes as a registered user using payment intents #{skip_confirm || ' not'} skipping confirm" do
-      create(
-        :stripe_payment_method,
-        preferred_stripe_intents_flow: 'payment',
-        preferred_skip_confirmation_for_payment_intent: skip_confirm,
-      )
-      visit_payment_step(user: create(:user))
-      choose_new_stripe_payment
-      fill_stripe_form
+        it 'successfully reuses a previously saved card from the user\'s wallet' do
```

Comment:
nit: if rubocop is not allowing to use double quotes when we have a single quote inside the string we should change the rule